### PR TITLE
feat: add across artifacts to contracts-node and contracts-frontend

### PR DESCRIPTION
**Pull request by mrice32**
**Motivation**

Contracts-node and contracts-frontend should support across without across smart contracts being included in the repo.

**Summary**

These packages now pull in across artifacts directly via the @across-protocol/contracts npm package.

Note: for now, this shouldn't really affect anything since these contracts _are_ included in the repo and haven't been removed (that will be the following step).

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [x]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

**Issue(s)**

N/A

### CODE REVIEW

**Review by chrismaree — APPROVED** [review]
this is pretty awesome! 

One question I have is as we add additional external packages that we want to be exported from `contracts-node` and `contracts-frontend` (say we make other projects similar to across that live outside of `protocol`) we'll need to make similar modifications to what you've done here. Do you see some pattern that can generalize what you've done here wherein we can simply provide a series of npm packages and it includes them within the `contracts-node` & `contracts-frontend` exports?

This is not something we need to worry about now, I was more just curious what your thinking is in how this scales over time.

**Review by nicholaspai — APPROVED** [review]

**Comment by mrice32** [comment]
> this is pretty awesome!
> 
> One question I have is as we add additional external packages that we want to be exported from `contracts-node` and `contracts-frontend` (say we make other projects similar to across that live outside of `protocol`) we'll need to make similar modifications to what you've done here. Do you see some pattern that can generalize what you've done here wherein we can simply provide a series of npm packages and it includes them within the `contracts-node` & `contracts-frontend` exports?
> 
> This is not something we need to worry about now, I was more just curious what your thinking is in how this scales over time.

@chrismaree I tried to make this as generalized as possible. I also added a section to the readme for contracts-node and cotnracts-frontend explaining how to add a new package.

File: packages/common/src/hardhat/tasks/artifacts.ts [hunk 1]
```diff
@@ -5,6 +5,12 @@ import { task, types } from "hardhat/config";
 import { HardhatRuntimeEnvironment } from "hardhat/types";
 import { CombinedHRE } from "./types";
 
+// Assumes these packages have the following structure:
```
Comment (chrismaree) [inline packages/common/src/hardhat/tasks/artifacts.ts:8]: this is much better than how you had it before as we can expand to `@across-protocol/contracts-v2` and all future projects that follow this separation pattern.

File: packages/common/src/hardhat/tasks/artifacts.ts [hunk 2]
```diff
@@ -163,32 +211,34 @@ task("generate-contracts-node", "Generate typescipt for the contracts-node packa
...
+          `export type { ${contractName} as ${contractName}Ethers } from "../typechain/${packageName}/ethers";\n`
+        );
     });
-    fs.appendFileSync(out, '} from "../typechain/ethers";\n');
 
-    fs.appendFileSync(out, "export {\n");
-    artifacts.forEach(({ contractName }) => {
-      if (fs.existsSync(`typechain/ethers/factories/${contractName}__factory.ts`))
-        fs.appendFileSync(out, `  ${contractName}__factory as ${contractName}Ethers__factory,\n`);
+    artifacts.forEach(({ contractName, packageName }) => {
```
Comment (chrismaree) [inline packages/common/src/hardhat/tasks/artifacts.ts:225]: sheesh. this looks like it was a pain to get exactly right.

File: packages/contracts-node/README.md [hunk 3]
```diff
@@ -99,3 +99,32 @@ await hre.deployments.deploy("Voting", { args: YOUR_ARGS, from: YOUR_ADDRESS });
...
+
+- All hardhat artifacts that it wants this package to re-export.
+- Typechain types under `/typechain`. It must include web3 and ethers typechain exports.
+- A special json file at the path `/build/artifacts.json` that describes the location of all hardhat artifacts. This
+  JSON file is a simple json array of objects, each object having a field called `relativePath` that is the path from the
+  root to each hardhat artifact. The artifact's filename must match the contract name.
+- A folder at `/networks` following the same structure as `/packages/core/networks` in this repository for any
+  contract addresses that it wants to add.
+
+### Adding an external package
```
Comment (chrismaree) [inline packages/contracts-node/README.md:120]: 👍

**Review by chrismaree — APPROVED** [review]
this is excellent! node wrangling like this is super painful. good work on generalising this process.

**Pull request merged**